apixu-weatherbot1.0.tcl

```diff
@@ -0,0 +1,498 @@
+        # conditional will be true and throw an error for illegal characters.
+        if {[regexp {[^\w., ]} $location] && ![regexp {^[a-zA-Z]+:[a-zA-Z]+$} $location]} {
+            putlog "weather::_set_location caught illegal characters"
+            error "Illegal characters in your location query. Acceptable characters:\
+                   a-z, 0-9, commas, and spaces. Colons only used i.e. 'iata:ksjc'"
+        }
+
+        if {!($units eq "0" || $units eq "1" || $units eq "2")} {
+            putlog "$nick set units to an invalid number."
+            error "Units must be specified from 0-2 where 0 = metric, 1 = imperial and 2\
+                   = both. i.e. \002'.set 2 New Orleans, LA'\002 would spam both unit\
+                   types for New Orleans."
+        } elseif {![string length $location]} {
+            putlog "$nick gave an invalid location"
+            error "Must supply a location to be set. i.e. \002'set 1\
+                    New Orleans, LA'\002"
+        } elseif {![validuser $hand]} {
+            adduser $nick
+            setuser $nick HOSTS [maskhost [getchanhost $nick] 
+            chattr $nick -hp
+            putlog "weather::set_location added user: $nick with host: $mask"
+        }
+
+        # Sets the location and units to each user
+        # in the eggdrop userfile.
+        setuser $hand XTRA weather.location $location
+        setuser $hand XTRA weather.units $units
+
+        switch $units {
+            "0" {
+                set units "metric"
+            }
+            "1" {
+                set units "imperial"
+            }
+            "2" {
+                set units "metric & imperial"
+            }
+        }
+
+        putlog  "$nick set their default location to $location."
+
+        set location_info [dict create location $location units $units]
+        return $location_info    
+    }
+
+    proc _get_help {nick} {
+        # Help section of the script called by user.
+        #
+        # Args:
+        #   nick(string) - Users nick.
+        #
+        # Returns:
+        #   (string) - Returns the list of commands and
+        #              description/examples of each one.
+        #
+        putlog "weather::_get_help called"
+        puthelp "PRIVMSG $nick :Commands:"
+        puthelp "PRIVMSG $nick :.wz <location> - Show current weather. <location> is\
+                optional depending if you have your weather set to bot."
+        puthelp "PRIVMSG $nick :.wzf <location> - Show a 5 day forecast. <location> is\
+                 optional."
+        puthelp "PRIVMSG $nick :.set 0-2 <location> - Set a default location. Units\
+                must be specified from 0-2 where 0 = metric, 1 = imperial and 2 = both.\
+                i.e. \".set 2 New Orleans, LA\" would spam both unit types for New Orleans."
+    }
+}
+
+putlog "apixu-weatherbot1.0.tcl (https://github.com/bolivierjr/apixu-weatherbot) loaded"
```

Comment:
maybe kill all the logging or put it in behind a method that flags on and off with a env var

Reply:
You want nothing to go in the logs? Kinda helpful to see errors/messages is it not?